src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0035Tests.cs

```diff
@@ -530,5 +530,87 @@ public class TestClient
 
             await Verifier.CreateAnalyzer(code).RunAsync();
         }
+
+        [Fact]
+        public async Task AZC0035_IgnoresDependencyTypes()
+        {
+            const string code = @"
+using Azure;
+using System.Threading.Tasks;
+
+namespace Azure.Test
+{
+    // This should be flagged because it's defined in current source
+    public class {|AZC0035:SourceDefinedModel|}
+    {
+        private SourceDefinedModel() { }
+        public string Name { get; }
+    }
+
+    public class TestClient
+    {
+        public virtual Response<SourceDefinedModel> GetSourceDefinedModel()
+        {
+            return null;
+        }
+
+        // These would be types from dependencies and should be ignored
+        // Note: In actual compilation, Azure.Response<T> and System.String are from dependencies
```

Comment:
Would `Azure.Response<T>` normally require a model factory?    Is that a reasonable test, or should we use another external dependency outside the `System` namespace?